cmake/modules/Finduuid.cmake

```diff
@@ -34,16 +34,18 @@ if(NOT UUID_INCLUDE_DIR)
   find_path(UUID_INCLUDE_DIR uuid/uuid.h)
 endif()
 
-if(EXISTS UUID_INCLUDE_DIR)
+if(EXISTS "${UUID_INCLUDE_DIR}")
   set(UUID_INCLUDE_DIRS ${UUID_INCLUDE_DIR})
   set(CMAKE_REQUIRED_INCLUDES ${UUID_INCLUDE_DIRS})
   check_cxx_symbol_exists("uuid_generate_random" "uuid/uuid.h" _uuid_header_only)
 endif()
 
-if(NOT _uuid_header_only AND NOT UUID_LIBRARY)
+if(NOT _uuid_header_only)
   check_library_exists("uuid" "uuid_generate_random" "" _have_libuuid)
   if(_have_libuuid)
-    set(UUID_LIBRARY "uuid")
+    if(NOT UUID_LIBRARY)
+      set(UUID_LIBRARY "uuid")
+    endif()
     set(UUID_LIBRARIES ${UUID_LIBRARY})
   endif()
 endif()
```

Comment:
Maybe we need an `else` clause here to set `UUID_LIBRARIES` to `UUID_LIBRARY` if that has been set by the user?

Reply:
Actually, looking again at these changes, the only problem may be that if the user sets both `UUID_INCLUDE_DIR` and `UUID_LIBRARY`, the latter may be ignored if it's decided that uuid is header only.